zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,870 @@
+blaze expression into a pipeline DataSet. This dataset will have a column for
+each field of the record. Some datashape types cannot be coerced into Pipeline
+API compatible types and in that case, a column cannot be constructed.
+Currently any numeric type that may be promoted to a float64 is compatible with
+the Pipeline API.
+
+The scalar case defines a single column pulled out a table. For example, let
+``expr = bz.symbol('s', 'var * {field: int32, asof_date: datetime}')``.
+When we pass ``expr.field`` to ``from_blaze``, we will walk back up the
+expression tree until we find the table that ``field`` is defined on. We will
+then proceed with the record case to construct a dataset; however, before
+returning the dataset we will pull out only the column that was passed in.
+
+For full documentation, see ``help(from_blaze)`` or ``from_blaze?`` in jupyter.
+
+Using our Pipeline DataSets and Columns
+---------------------------------------
+
+Once we have mapped our blaze expressions into Pipeline API objects, we may
+use them just like any other datasets or columns. For more information on how
+to run a pipeline or using the Pipeline API, see:
+www.quantopian.com/help#pipeline-api
+"""
+from __future__ import division, absolute_import
+
+from abc import ABCMeta, abstractproperty
+from collections import namedtuple, defaultdict
+from itertools import count
+import warnings
+from weakref import WeakKeyDictionary
+
+import blaze as bz
+from datashape import (
+    Date,
+    DateTime,
+    Option,
+    float64,
+    isrecord,
+    isscalar,
+    promote,
+)
+from numpy.lib.stride_tricks import as_strided
+from odo import odo
+import pandas as pd
+from toolz import (
+    complement,
+    compose,
+    concat,
+    flip,
+    groupby,
+    identity,
+    memoize,
+)
+import toolz.curried.operator as op
+from six import with_metaclass, PY2, iteritems
+
+
+from ..data.dataset import DataSet, Column
+from zipline.lib.adjusted_array import adjusted_array
+from zipline.lib.adjustment import Float64Overwrite
+
+
+AD_FIELD_NAME = 'asof_date'
+TS_FIELD_NAME = 'timestamp'
+SID_FIELD_NAME = 'sid'
+valid_deltas_node_types = (
+    bz.expr.Field,
+    bz.expr.ReLabel,
+    bz.expr.Symbol,
+)
+traversable_nodes = (
+    bz.expr.Field,
+    bz.expr.Label,
+)
+is_invalid_deltas_node = complement(flip(isinstance, valid_deltas_node_types))
+getname = op.attrgetter('__name__')
+
+
+class _ExprRepr(object):
+    """Box for repring expressions with the str of the expression.
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression to box for repring.
+    """
+    __slots__ = 'expr',
+
+    def __init__(self, expr):
+        self.expr = expr
+
+    def __repr__(self):
+        return str(self.expr)
+    __str__ = __repr__
+
+
+class ExprData(namedtuple('ExprData', 'expr deltas resources')):
+    """A pair of expressions and a data resources.
+
+    Parameters
+    ----------
+    expr : Expr
```

Comment:
I've referred to this concept as "baseline" elsewhere in the pipeline infrastructure.